File: src/main/java/com/giant/auth/repository/custom/AccountRepositoryCustom.java
package com.giant.auth.repository.custom;

import com.giant.auth.dto.SignInDto;

import java.util.Optional;

public interface AccountRepositoryCustom {
    Optional<SignInDto> findSignInInfoByUserName(String userName);
    Optional<SignInDto> findRefreshInfoByAccountId(Long accountId);
}

File: src/main/kotlin/com/giant/auth/repository/custom/AccountRepositoryCustom.kt
package com.giant.auth.repository.custom

import com.giant.auth.dto.SignInDto

interface AccountRepositoryCustom {
    fun findSignInInfoByUserName(userName: String): SignInDto?
    fun findRefreshInfoByAccountId(accountId: Long): SignInDto?
    fun existsByUserName(userName: String): Boolean?
    fun findPasswordHashByAccountId(accountId: Long): String?
    fun updateAccount(accountId: Long, userName: String, phoneNumber: String, email: String): Long
    fun updatePassword(accountId: Long, passwordHash: String): Long
}